src/EditorFeatures/CSharpTest/AssignOutParameters/AssignOutParametersTests.cs

```diff
@@ -0,0 +1,534 @@
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestNestedReturnNoBlock()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i)
+    {
+        if (true)
+            [|return 'a';|]
+    }
+}",
+@"class C
+{
+    char M(out int i)
+    {
+        if (true)
+        {
+            i = 0;
+            return 'a';
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestNestedReturnEvenWhenWrittenAfter()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    char M(bool b, out int i)
+    {
+        if (b)
+        {
+            [|return 'a';|]
+        }
+
+        i = 1;
+    }
+}",
+@"class C
+{
+    char M(bool b, out int i)
+    {
+        if (b)
+        {
+            i = 0;
+            return 'a';
+        }
+
+        i = 1;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestForExpressionBodyMember()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i) => [|'a';|]
+}",
+@"class C
+{
+    char M(out int i)
+    {
+        i = 0;
+        return 'a';
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestForLambdaExpressionBody()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    delegate char D(out int i);
+    void X()
+    {
+        D d = (out int i) => [|'a';|]
+    }
+}",
+@"class C
+{
+    delegate char D(out int i);
+    void X()
+    {
+        D d = (out int i) => { i = 0; return 'a'; };
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestMissingForLocalFunctionExpressionBody()
+        {
+            // Note desirable.  Happens because error is not reported on expr-body
+            // like for lambdas.
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void X()
+    {
+        char D(out int i) => [|'a';|]
+        D(out _);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestForLambdaBlockBody()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    delegate char D(out int i);
+    void X()
+    {
+        D d = (out int i) =>
+        {
+            [|return 'a';|]
+        }
+    }
+}",
+@"class C
+{
+    delegate char D(out int i);
+    void X()
+    {
+        D d = (out int i) =>
+        {
+            i = 0;
+            return 'a';
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestForLocalFunctionBlockBody()
```

Comment:
Not sure if you already have a test for:

```csharp
void M(out int i)
{
    Initialize();
    return;

    void Initialize() => i = 0;
}
```

Will this figure out that the invoked local function assigns the out param on all paths?

Reply:
Local function cannot touch out parameters iirc.

Reply:
indeed.  won't fix.